src/main/java/interface_adaptors/gatewayReader.java

```diff
@@ -0,0 +1,31 @@
+package interface_adaptors;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class gatewayReader {
+
+    private final File file;
+
+    public gatewayReader(String filePath){
+        this.file = new File(filePath);
+    }
+
+    public gatewayReader(File file){
+        this.file = file;
+    }
+    public List<String[]> getData() throws IOException {
+        BufferedReader reader = new BufferedReader(new FileReader(file));
+        List<String[]> fileData = new ArrayList<>();
```

Comment:
I guess this fileData is the list of string that contains each line of the data in the file (which means each line of the data would stored as each element in fileData).